### DISCUSSION TITLE
Not handling some events using `on` method, but works fine with `onAny` method.

### DISCUSSION THREAD

**Discussion by moltar** [socketio/socket.io#5228] [General]
```ts
  const webSocketClient = getWebSocketClient()

  webSocketClient.removeAllListeners('module_update')

  console.log('b4 module_update')
  webSocketClient.on('module_update', () => {
    console.log('in module_update')
    moduleStore.refreshSummary()
  })

  webSocketClient.onAny((eventName, ...args) => {
    console.log('on any', eventName, args)
  })
```

I see `b4 module_update` printed.

When events get emitted, I see `module_update` being caught by `.onAny`, but not by the `.on` method.

![screenshot-20241114T204530-wJ71YOeO@2x](https://github.com/user-attachments/assets/4c891016-6de1-4dc0-b539-a63e41e09e1d)

What gives?

Socket instance is created like this:

```ts
    SOCKET = io('', {
      path: `${VITE_WEBSOCKET_ENDPOINT}/socket.io/`,
      transports: ['websocket'],
      autoConnect: false,
    })
```

Then I call `webSocketClient.connect()` manually, later, when user has passed the auth.

Any clues?

**Comment by darrachequesne** [comment]
Hi! I was not able to reproduce the issue locally. Have you found the culprit?

**Reply by moltar** [reply]
No, did not find. I think it's a race condition. But I looked at the Socket.io code, and it is certainly handling `on` and `onAny` differently, at least it's in different places/classes.
